Objects/genobject.c

```diff
@@ -1951,6 +1952,7 @@ async_gen_wrapped_val_dealloc(_PyAsyncGenWrappedValue *o)
 #endif
     if (state->value_numfree < _PyAsyncGen_MAXFREELIST) {
         assert(_PyAsyncGenWrappedValue_CheckExact(o));
+        _PyGC_CLEAR_FINALIZED((PyObject *)o);
```

Comment:
Oh clicked wrong button, you need to call this in async_gen_asend_dealloc on `asend_freelist`

Reply:
Oops. I fixed the location of `_PyGC_CLEAR_FINALIZED` and updated `test_asend` in `test_asyncgen.py` so that it would catch this.